src/plugins/discover/public/application/angular/discover.js

```diff
@@ -199,6 +201,14 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
   const savedSearch = $route.current.locals.savedObjects.savedSearch;
   $scope.searchSource = savedSearch.searchSource;
   $scope.indexPattern = resolveIndexPatternLoading();
+  $scope.useNewFieldsApi = !$scope.searchSource.dependencies?.getConfig(
+    UI_SETTINGS.SEARCH_FIELDS_FROM_SOURCE
+  );
+  $scope.showUnmappedFields = false;
+  $scope.onShowUnmappedFieldsChange = (val) => {
+    $scope.showUnmappedFields = val;
```

Comment:
This is implemented to propagate until here, but it doesn't really do anything yet.

Reply:
Would we need another request to get those fields?

Reply:
Yes :D